extensions/simon/src/index.tsx

```diff
@@ -0,0 +1,193 @@
+          }
+
+          return colour;
+        }),
+      }));
+    }, 500);
+  };
+
+  const animateSequence = (newSequence: string[]) => {
+    setState((previous) => ({ ...previous, loading: true }));
+
+    newSequence.forEach((colour, index) => {
+      setTimeout(() => {
+        activateColour(colour);
+      }, (index + 1) * 800);
+    });
+
+    setState((previous) => ({ ...previous, loading: false }));
+  };
+
+  useEffect(() => {
+    if (state.gameState === "play") {
+      if (state.humanSequence.length === state.sequence.length) {
+        if (state.humanSequence.join("") === state.sequence.join("")) {
+          if (state.level < maxLevel) {
+            setState((previous) => ({
+              ...previous,
+              level: state.level + 1,
+              humanSequence: [],
+            }));
+
+            const nextColour = nextLevel();
+
+            animateSequence([...state.sequence, nextColour]);
+          } else {
+            exec("open raycast://confetti");
+
+            setState((previous) => ({ ...previous, gameState: "win" }));
+          }
+        } else {
+          setState((previous) => ({ ...previous, gameState: "lose" }));
+        }
+      }
+    }
+  }, [state.gameState, state.humanSequence]);
+
+  if (state.gameState === "lobby") {
+    return (
+      <Detail
+        markdown="Press Start Game to start playing!"
+        actions={
+          <ActionPanel>
+            <Action.SubmitForm
+              icon={Icon.Star}
+              title="Start Game"
+              onSubmit={() => {
+                setState((previous) => ({ ...previous, gameState: "play" }));
+
+                const nextColour = nextLevel();
+
+                activateColour(nextColour);
+
+                animateSequence(state.sequence);
+              }}
+            />
+          </ActionPanel>
+        }
+      />
+    );
+  }
+
+  if (state.gameState === "win" || state.gameState === "lose") {
+    return (
+      <Detail
+        markdown={state.gameState === "win" ? "You won, congratulations!" : "You lost, better luck next time!"}
+        actions={
+          <ActionPanel>
+            <Action.SubmitForm
+              icon={Icon.Star}
+              title="Play Again"
+              onSubmit={() => {
+                setState((previous) => ({
+                  ...previous,
+                  loading: false,
+                  gameState: "lobby",
+                  level: 1,
+                  sequence: [],
+                  humanSequence: [],
+                }));
+              }}
+            />
+          </ActionPanel>
+        }
+      />
+    );
+  }
+
+  return (
+    <Grid itemSize={Grid.ItemSize.Medium} inset={Grid.Inset.Small} isLoading={state.loading}>
+      {state.colours.map((colour, index) => (
+        <Grid.Item
+          key={colour.name}
+          content={{ value: { source: Icon.Circle, tintColor: colour.tint }, tooltip: colour.name }}
```

Comment:
Not sure that tooltip makes sense?

Reply:
I believe the tooltip is required when using an icon like this?